src/nvim/mark.c

```diff
@@ -378,8 +378,8 @@ pos_T *getmark_buf_fnum(buf_T *buf, int c, int changefile, int *fnum)
     }
   } else if (ASCII_ISLOWER(c)) {      /* normal named mark */
     posp = &(buf->b_namedm[c - 'a']);
-  } else if (ASCII_ISUPPER(c) || VIM_ISDIGIT(c)) {    /* named file mark */
-    if (VIM_ISDIGIT(c))
+  } else if (ASCII_ISUPPER(c) || isdigit(c)) {    /* named file mark */
```

Comment:
> isdigit(c) is checked twice if "c" is a digit

It looks like there would be a lot of duplication if they were separated out like this:

```
...

} else if (ASCII_ISUPPER(c)) {

...

} else if (isdigit(c)) {

...

```

A `goto` could be used, but the current redundancy is so small I don't really care.

> also a manual subtraction is used instead of something like "tolower()"

I've noticed that in a few places in Nvim. Vim works on some platforms that (at least according to their comments) have some pretty standard libraries, so It'd probably be worthwhile to deal with that given the platforms we support are relatively sane.